File: pset6/caesar.py
#!/usr/bin/env python3

import cs50
import sys

def main():
    
    if len(sys.argv) != 2:
        print('caesar.py k')
        exit(1)
        
    k = int(sys.argv[1])
    
    print('plaintext: ', end = '')
    
    s = cs50.get_string()
    
    print('ciphertext: ', end = '')
    
    for i in range(len(s)):
        
        c = s[i]
        
        if str.isupper(c):
            cipher = (((ord(c) - 65) + k) % 26) + 65
            print(chr(cipher), end = '')
        elif str.islower(c):
            cipher = (((ord(c)- 97) + k) % 26) + 97
            print(chr(cipher), end = '')
        else:
            print(c, end = '')
            
    
    print('')
    
    exit(0)
            
    
if __name__ == "__main__":
    main()

File: pset2/caesar/caesar.c
#include <stdio.h>
#include <stdlib.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
 
int main(int argc, string argv[])
{	
    if (argc != 2)
    {
        printf("./caesar k\n");
        return 1;
    }
 
    int k = atoi(argv[1]);
    
    printf("plaintext: ");
    
    string s = GetString();
    
    printf("ciphertext: ");
 
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        char c = s[i];
 
        if (isupper(c))
        {   
            char cipher = (((c - 65) + k) % 26) + 65;
            printf("%c", cipher);
        }
        else if(islower(c))
        {
            char cipher = (((c - 97) + k) % 26) + 97;
            printf("%c", cipher);
        }
        else
        {
            printf("%c", c);
        }
    }
    printf("\n");
    
    return 0;
}